src/design-patterns/MazeGame/maze.cpp

```diff
@@ -0,0 +1,24 @@
+#include "maze.hpp"
+
+#include <iostream>
+
+Maze::Maze() { std::cout << "Init Maze()" << std::endl; }
+
+void Maze::AddRoom(std::unique_ptr<Room> room) {
+  int number = room->GetRoomNumber();
+  rooms.push_back(std::move(room));
```

Comment:
이동생성자가 있다면 아래와 같이 쓸 수 있음 
```suggestion
  rooms.emplace_back (room);
```